bigquery_etl/cli/query.py

```diff
@@ -0,0 +1,144 @@
+"""bigquery-etl CLI query command."""
+
+import click
+import os
+from pathlib import Path
+import re
+import sys
+import string
+
+from bigquery_etl.metadata.parse_metadata import Metadata
+
+
+QUERY_NAME_RE = re.compile(
+    r"(?P<dataset>[a-zA-z0-9_]+)\.(?P<name>[a-zA-z0-9_]+)(?P<ver>_v[0-9]+)?"
+)
+
+
+def create_query_cli_group():
+    """Create the CLI group for the query command."""
+    group = click.Group(name="query", help="Creating/validating/deleting/... queries")
+    group.add_command(query_create_command)
+    return group
+
+
+@click.command(
+    name="create",
+    help=(
+        "Create a new query with name "
+        "<dataset>.<query_name>, for example: telemetry_derived.asn_aggregates"
+    ),
+)
+@click.argument("name")
+@click.option(
+    "--path",
+    "-p",
+    help="Path to directory in which query should be created",
+    default="sql/",
+)
+@click.option(
+    "--owner",
+    "-o",
+    help="Owner of the query (email address)",
+    default="example@mozilla.com",
+)
+@click.option(
+    "--init",
+    "-i",
+    help="Create an init.sql file to initialize the table",
+    default=False,
+)
+def query_create_command(name, path, owner, init):
+    """CLI command for creating a new query."""
+    if not os.path.isdir(path):
+        click.echo(f"Invalid path for adding new query: {path}", err=True)
+        sys.exit(1)
+
+    # create directory structure for query
+    try:
+        match = QUERY_NAME_RE.match(name)
+        name = match.group("name")
+        dataset = match.group("dataset")
+        version = match.group("ver") or "_v1"
+    except AttributeError:
+        click.echo(
+            "New queries must be named like:"
+            + " <dataset>.<table> or <dataset>.<table>_v[n]"
+        )
+        sys.exit(1)
+
+    derived_path = None
+    view_path = None
+
+    if dataset.endswith("_derived"):
+        # create a directory for the corresponding view
+        derived_path = Path(path) / dataset / (name + version)
+        os.makedirs(derived_path)
```

Comment:
Prefer working within pathlib over os if you can, for consistency. https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir

```suggestion
        derived_path.mkdir(parents=True)
```